docs/docs/policy-performance.md

```diff
@@ -968,6 +968,125 @@ This feature can be enabled for `opa run`, `opa eval`, and `opa bench` by settin
 
 Users are recommended to do performance testing to determine the optimal configuration for their use case.
 
+## Performance Metrics
+
+OPA exposes metrics for each phase of policy evaluation:
+
+- **System-wide metrics** are available at the `/metrics` Prometheus endpoint
+- **Per-query metrics** are returned with individual API responses when `?metrics=true` is specified
+
+See [Monitoring](./monitoring#metrics-overview) for the distinction between these metric types.
+
+### Query Evaluation Metrics
+
+Query evaluation phases:
+
+- `timer_rego_query_parse_ns` - Time spent parsing the query string into AST
+- `timer_rego_query_compile_ns` - Time spent compiling the query for evaluation
+- `timer_rego_query_eval_ns` - Time spent executing the compiled query
+
+Compilation time often dominates in complex policies.
+
+### Module and Policy Metrics
+
+Policy compilation and parsing:
+
+- `timer_rego_module_parse_ns` - Time to parse policy modules from source
+- `timer_rego_module_compile_ns` - Time to compile parsed modules into evaluation form
+- `timer_rego_data_parse_ns` - Time to parse data documents
+- `timer_rego_input_parse_ns` - Time to parse input documents
+
+Module compilation runs once at load time. Slow compilation impacts bundle updates.
+
+### File and Bundle Loading
+
+Policy and data loading:
+
+- `timer_rego_load_files_ns` - Time to load policy files from disk
+- `timer_rego_load_bundles_ns` - Time to load and activate bundles
+- `timer_bundle_request_ns` - Time spent downloading bundles
```

Comment:
Some of these are tricky to categorise. But I think it's not ideal to have this one in 'policy performance'.